templates/cloud_set_guest_password.j2

```diff
@@ -0,0 +1,125 @@
+
+# Modify this line to specify the user (default is root)
+user=root
+
+# Add your DHCP lease folders here
+DHCP_FOLDERS="/var/lib/dhclient/* /var/lib/dhcp3/* /var/lib/dhcp/* /var/db/dhclient.leases.*"
+PASSWORD_SERVER_PORT=8080
+password_received=0
+file_count=0
+error_count=0
+
+for DHCP_FILE in $DHCP_FOLDERS; do
+	if [ -f $DHCP_FILE ]; then
+		file_count=$((file_count+1))
+		PASSWORD_SERVER_IP=$(grep dhcp-server-identifier $DHCP_FILE | tail -1 | awk '{print $NF}' | tr -d '\;')
+
+		if [ -n "$PASSWORD_SERVER_IP" ]; then
+			logger -t "cloud" "Found password server IP $PASSWORD_SERVER_IP in $DHCP_FILE"
+                        break
+		fi
+	fi
+done
+
+if [ -z "$PASSWORD_SERVER_IP" ] ; then
+         logger -t "cloud" "Unable to determine the password server, falling back to data-server"
+         PASSWORD_SERVER_IP=data-server
+fi
+
+logger -t "cloud" "Sending request to password server at $PASSWORD_SERVER_IP"
+password=$(wget -q -t 3 -T 20 -O - --header "DomU_Request: send_my_password" http://$PASSWORD_SERVER_IP:$PASSWORD_SERVER_PORT)
+if [ $? -eq 0 ]; then
+        password=$(echo $password | tr -d '\r')
+        logger -t "cloud" "Got response from server at $PASSWORD_SERVER_IP"
+
+        case $password in
+                "")
+                        logger -t "cloud" "Password server at $PASSWORD_SERVER_IP did not have any password for the VM"
+                        ;;
+                "bad_request")
+                        logger -t "cloud" "VM sent an invalid request to password server at $PASSWORD_SERVER_IP"
+                        error_count=$((error_count+1))
+                        ;;
+                "saved_password")
+                        logger -t "cloud" "VM has already saved a password from the password server at $PASSWORD_SERVER_IP"
+                        ;;
+                *)
+                        logger -t "cloud" "VM got a valid password from server at $PASSWORD_SERVER_IP"
+                        password_received=1
+                        ;;
+                esac
+else
+        logger -t "cloud" "Failed to send request to password server at $PASSWORD_SERVER_IP"
+        error_count=$((error_count+1))
+fi
+
+
+if [ "$password_received" == "0" ]; then
+	if [ "$error_count" == "$file_count" ]; then
+		logger -t "cloud" "Failed to get password from any server"
+		exit 1
+	else
+		logger -t "cloud" "Did not need to change password."
+		exit 0
+	fi
+fi
+
+logger -t "cloud" "Changing password for user $user"
+
+{% if ansible_os_family == 'FreeBSD' %}
+echo "${password}" | /usr/sbin/pw usermod -n ${user} -h 0 2>/dev/null
+{% elif ansible_os_family == 'OpenBSD' %}
+encrypted=$(echo "${password}" | encrypt -b 6)
+usermod -p "${encrypted}" ${user}
+{% else %}
+echo $user:$password | chpasswd
+{% endif %}
+exit_status=$?
+
+if [ $exit_status -gt 0 ]; then
+{% if ansible_os_family == 'FreeBSD' %}
+    echo "${password}" | /usr/sbin/pw usermod -n ${user} -h 0 2>/dev/null
```

Comment:
it seems the `if` block in the original code is for falling back to `usermod` when `chpasswd` fails.
repeating the same command in FreeBSD and OpenBSD is meaningless.